presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonPath.java

```diff
@@ -13,19 +13,142 @@
  */
 package com.facebook.presto.operator.scalar;
 
+import com.facebook.presto.common.function.JsonPathExtractionEngine;
+import com.facebook.presto.spi.PrestoException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.InvalidJsonException;
+import com.jayway.jsonpath.InvalidPathException;
+import com.jayway.jsonpath.PathNotFoundException;
+import com.jayway.jsonpath.spi.json.JacksonJsonNodeJsonProvider;
 import io.airlift.slice.Slice;
 
+import java.io.IOException;
+import java.io.InputStream;
+
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static io.airlift.slice.Slices.utf8Slice;
+import static java.lang.String.format;
+
 public class JsonPath
 {
     private final JsonExtract.JsonExtractor<Slice> scalarExtractor;
     private final JsonExtract.JsonExtractor<Slice> objectExtractor;
     private final JsonExtract.JsonExtractor<Long> sizeExtractor;
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    private static JsonExtract.JsonExtractor<Slice> getScalarExtractorForJayway(com.jayway.jsonpath.JsonPath jsonPath, Configuration jaywayConfig)
+    {
+        return new JsonExtract.JsonExtractor<Slice>()
+        {
+            @Override
+            public Slice extract(InputStream inputStream)
+                    throws IOException
+            {
+                JsonNode node = jaywayExtract(jsonPath, jaywayConfig, inputStream);
+                if (node == null || !node.isValueNode()) {
+                    return null;
+                }
+                return utf8Slice(node.asText());
```

Comment:
why asText here and toString in other place ?

Reply:
toString will return a json string representation of the node, e.g. if the node is a TextNode then toString will return `"\"foo\""`, while asText returns a String representation of the contained scalar value (`"foo"`).
So for json_extract_scalar I believe asText is what we want.